### DISCUSSION TITLE
Adding Own SSL Certificates

### DISCUSSION THREAD

**Discussion by BenjaminM007** [btcpayserver/btcpayserver#6110] [Technical Support]
Hello

We have deployed BTC pay server months ago.
We use Cloudflare so the lets encrypt cert does not renew.

How and where to we change the lets encrypt ssl cert to our own so that we do not reply upon the lets encrypt ssl certificate.?

**Comment by Rklosnet** [comment]
Sorry I can't help you, but maybe you can help us. How did you manage to let BTCPay communicate with your website after using Cloudflare? Ever since we use Cloudflare, we have to manually complete all our orders. Incredibly frustrating. We are using Lunanode to host our BTCPayserver

**Reply by BenjaminM007** [reply]
You need to create page rules ( which are now depreciated ) to whitelist the callback URL to your server.
We use our BTC pay server on a private network so we are not using the callback via the internet.

You should try whitelisting the lunanode IP address in cloudflare and ensure you create page rules for the callback URL. Ensure WAF is disabled in the page rule and performance as these will block the callback.

**Reply by Rklosnet** [reply]
Kind of you to help sir, I hope your questions will get answered shortly!

**Comment by AdilElFarissi** [comment]
I add my question mark to this question... I have apache (+certbot) which uses ports 80 / 443 and BTCPay on other sub-domain and ports... it would be good if BTCPay was able to use certificates from outside docker...
Thank you :)

**Reply by BenjaminM007** [reply]
Completely agree. This is needed.
